impl_core/src/main/java/io/opencensus/implcore/trace/propagation/CloudTraceFormat.java

```diff
@@ -0,0 +1,142 @@
+ * </ul>
+ */
+final class CloudTraceFormat extends TextFormat {
+  static final String HEADER_NAME = "X-Cloud-Trace-Context";
+  static final List<String> FIELDS = Collections.singletonList(HEADER_NAME);
+  static final char SPAN_ID_DELIMITER = '/';
+  static final String TRACE_OPTION_DELIMITER = ";o=";
+  static final String SAMPLED = "1";
+  static final String NOT_SAMPLED = "0";
+  static final TraceOptions OPTIONS_SAMPLED = TraceOptions.builder().setIsSampled(true).build();
+  static final TraceOptions OPTIONS_NOT_SAMPLED = TraceOptions.DEFAULT;
+  static final int TRACE_ID_SIZE = 2 * TraceId.SIZE;
+  static final int TRACE_OPTION_DELIMITER_SIZE = TRACE_OPTION_DELIMITER.length();
+  static final int SPAN_ID_START_POS = TRACE_ID_SIZE + 1;
+  // 32-digit TRACE_ID + 1 digit SPAN_ID_DELIMITER + at least 1 digit SPAN_ID
+  static final int MIN_HEADER_SIZE = SPAN_ID_START_POS + 1;
+
+  @Override
+  public List<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C> void inject(SpanContext spanContext, C carrier, Setter<C> setter) {
+    checkNotNull(spanContext, "spanContext");
+    checkNotNull(setter, "setter");
+    checkNotNull(carrier, "carrier");
+    StringBuilder builder =
+        new StringBuilder()
+            .append(spanContext.getTraceId().toLowerBase16())
+            .append(SPAN_ID_DELIMITER)
+            .append(UnsignedLongs.toString(spanIdToLong(spanContext.getSpanId())))
+            .append(TRACE_OPTION_DELIMITER)
+            .append(spanContext.getTraceOptions().isSampled() ? SAMPLED : NOT_SAMPLED);
+
+    setter.put(carrier, HEADER_NAME, builder.toString());
+  }
+
+  @Override
+  public <C> SpanContext extract(C carrier, Getter<C> getter) throws SpanContextParseException {
+    checkNotNull(carrier, "carrier");
+    checkNotNull(getter, "getter");
+    try {
+      String headerStr = getter.get(carrier, HEADER_NAME);
+      if (headerStr == null || headerStr.length() < MIN_HEADER_SIZE) {
+        throw new SpanContextParseException("Missing or too short header: " + HEADER_NAME);
+      }
+      checkArgument(headerStr.charAt(TRACE_ID_SIZE) == SPAN_ID_DELIMITER, "Invalid TRACE_ID size");
+
+      TraceId traceId = TraceId.fromLowerBase16(headerStr.subSequence(0, TRACE_ID_SIZE));
+      int traceOptionsPos = headerStr.indexOf(TRACE_OPTION_DELIMITER, SPAN_ID_DELIMITER);
+      CharSequence spanIdStr =
+          headerStr.subSequence(
+              SPAN_ID_START_POS, traceOptionsPos < 0 ? headerStr.length() : traceOptionsPos);
+      SpanId spanId = longToSpanId(UnsignedLongs.parseUnsignedLong(spanIdStr.toString(), 10));
+      TraceOptions traceOptions = OPTIONS_NOT_SAMPLED;
+      if (traceOptionsPos > 0) {
+        String traceOptionsStr = headerStr.substring(traceOptionsPos + TRACE_OPTION_DELIMITER_SIZE);
+        if (traceOptionsStr.isEmpty()) {
+          throw new SpanContextParseException("Invalid TRACE_OPTIONS");
+        }
+        if (SAMPLED.equals(traceOptionsStr)) {
```

Comment:
I think this is one thing that is wrong here. This can be int32 and the last bit is the sample bit. you need to read as an int and compare the last bit. I think that's what the format says, please double check.

Reply:
Yes, I should have implemented it using bit compare. 
Just one more question, the spec defines it as a hex value, but the examples and the C++ implementation both treat it as a uint32 decimal. Which one is the latest?